### PR TITLE
Add quick-store

### PR DESCRIPTION
**Pull request by promise**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
I'd say most of the code here follows general guidelines for JS/TS, so interfaces and types names should be PascalCase. It's not enforced here though, but I'd say, it's a good practice that blends seamlessly with others code:
https://google.github.io/styleguide/tsguide.html#identifiers

File: types/quick-store/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Type definitions for quick-store 0.1
+// Project: https://www.npmjs.com/package/quick-store (repository is deleted)
+// Definitions by: Glenn <https://github.com/promise>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = QuickStore;
+
+interface dbContents { [key: string]: value; }
+type value = string | number | dbContents | value[] | boolean | null;
+type callback<value> = (data: value) => any;
```
Comment (peterblazejewicz) [inline types/quick-store/index.d.ts:10]:
```suggestion
type callback<value> = (data: value) => void;
```
we don't have source, but those are callbacks, not predicates or reducers. There is no notion on README those can be used in any other way (like returning a bool value to impact outcome of some action, etc)

File: types/quick-store/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+declare function QuickStore(filename?: string, data?: dbContents): QuickStore.Database;
+
+declare namespace QuickStore {
+    interface Database {
+        put(data: dbContents, callback?: callback<dbContents>): void;
+        setItem(key: string, value: value, callback?: callback<dbContents>): void;
+        getItem(key: string, callback: callback<value>): void;
+        removeItem(key: string, callback?: callback<dbContents>): void;
+        clear(callback?: callback<dbContents>): void;
+        get<optionalCallback extends callback<dbContents> | undefined>(callback?: optionalCallback): callbackExists<optionalCallback, void, dbContents>;
```
Comment (peterblazejewicz) [inline types/quick-store/index.d.ts:22]: you can use methods overloading  here I think, to denote different outcome. The rule is more specific on the top:
```ts
get(callback: CallbackType...): void; 
get(): Value
```
it's much cleaner to read and grasp that method actually has two different paths.
I can see usage of using TS features to actually create outcome type for return, but that usually make more sense in much more complicated declaration

Comment (promise) [reply types/quick-store/index.d.ts:22]: I tried doing this at first but got errors. Seems like it's working now, so probably something I did wrong yesterday. Thank you 👍

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@promise thx!

**Comment by promise** [comment]
Ready to merge

**Pull request merged**